src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioErrorReportingService.cs

```diff
@@ -161,102 +148,23 @@ public void OnClosed(IVsInfoBarUIElement infoBarUIElement)
         }
 
-        private void CreateInfoBarForCodeFix(IVsInfoBarUIFactory factory, IVsWindowFrame frame, string message, Action onClose, Action onEnable = null, Action onEnableAndIgnore = null)
+        private static bool TryCreateInfoBarUI(IVsInfoBarUIFactory infoBarUIFactory, IVsInfoBar infoBar, out IVsInfoBarUIElement uiElement)
         {
-            object unknown;
-            if (ErrorHandler.Failed(frame.GetProperty((int)__VSFPROPID7.VSFPROPID_InfoBarHost, out unknown)))
-            {
-                return;
-            }
-
-            var textSpans = new List<IVsInfoBarTextSpan>()
-            {
-                new InfoBarTextSpan(message)
-            };
-
-            // create action item list
-            var actionItems = new List<IVsInfoBarActionItem>();
-            if (onEnable != null)
-            {
-                actionItems.Add(s_enableItem);
-            }
-
-            if (onEnableAndIgnore != null)
-            {
-                actionItems.Add(s_enableAndIgnoreItem);
-            }
-
-            var infoBarModel = new InfoBarModel(
-                textSpans,
-                actionItems.ToArray(),
-                KnownMonikers.StatusInformation,
-                isCloseButtonVisible: true);
-
-            IVsInfoBarUIElement infoBarUI;
-            if (!TryCreateInfoBarUI(factory, infoBarModel, out infoBarUI))
-            {
-                return;
-            }
-
-            uint? infoBarCookie = null;
-            var eventSink = new CodeFixInfoBarEvents(() =>
-            {
-                onClose();
-
-                if (infoBarCookie.HasValue)
-                {
-                    infoBarUI.Unadvise(infoBarCookie.Value);
-                }
-            }, onEnable, onEnableAndIgnore);
-
-            uint cookie;
-            infoBarUI.Advise(eventSink, out cookie);
-            infoBarCookie = cookie;
-
-            IVsInfoBarHost host = (IVsInfoBarHost)unknown;
-            host.AddInfoBar(infoBarUI);
+            uiElement = infoBarUIFactory.CreateInfoBar(infoBar);
+            return uiElement != null;
         }
 
-        private class CodeFixInfoBarEvents : IVsInfoBarUIEvents
+        public void ShowDetailedErrorInfo(Exception exception)
         {
-            private readonly Action _onClose;
-            private readonly Action _onEnable;
-            private readonly Action _onEnableAndIgnore;
-
-            public CodeFixInfoBarEvents(Action onClose, Action onEnable = null, Action onEnableAndIgnore = null)
-            {
-                Contract.ThrowIfNull(onClose);
+            string errorInfo = exception.Message + Environment.NewLine + exception.StackTrace;
 
-                _onClose = onClose;
-                _onEnable = onEnable;
-                _onEnableAndIgnore = onEnableAndIgnore;
-            }
-
-            public void OnActionItemClicked(IVsInfoBarUIElement infoBarUIElement, IVsInfoBarActionItem actionItem)
+            while (exception.InnerException != null)
```

Comment:
Consider unwrapping `AggregateException` which might have multiple inner exceptions at the same level.